inc/commonitiltask.class.php

```diff
@@ -1970,4 +1983,126 @@ static function showVeryShort($ID, $itemtype) {
       }
    }
 
+   public static function getGroupItemsAsVCalendars($groups_id) {
+
+      return self::getItemsAsVCalendars([static::getTableField('groups_id_tech') => $groups_id]);
+   }
+
+   public static function getUserItemsAsVCalendars($users_id) {
+
+      return self::getItemsAsVCalendars([static::getTableField('users_id_tech') => $users_id]);
+   }
+
+   /**
+    * Returns items as VCalendar objects.
+    *
+    * @param array $criteria
+    *
+    * @return \Sabre\VObject\Component\VCalendar[]
+    */
+   private static function getItemsAsVCalendars(array $criteria) {
+
+      global $DB;
+
+      $item = new static();
+      $parent_item = getItemForItemtype($item->getItilObjectItemType());
+      if (!$parent_item) {
+         return;
+      }
+
+      $query = [
+         'SELECT'     => [$item->getTableField('*')],
+         'FROM'       => $item->getTable(),
+         'INNER JOIN' => [],
```

Comment:
Does not an empty inner join cause issues?

Reply:
Nope, it executes a foreach on an empty array, resulting in appending an empty string to the query.